src/vivarium_public_health/dataset_manager/artifact.py

```diff
@@ -33,9 +34,8 @@ def __init__(self, path: str, filter_terms: List[str]=None):
 
         self.path = path
         self._filter_terms = filter_terms
-
         self._cache = {}
-        self._keys = [EntityKey(k) for k in hdf.get_keys(self.path)]
+        self._keys = [EntityKey(k) for k in hdf.load(path, EntityKey('metadata.keyspace'), None)]
```

Comment:
We should think very carefully about encapsulation here.  Before we introduced the `metadata` keys, we didn't actually have hard requirements on what was in the artifact.  This update means we have to create a new hdf file, then also put in an empty list at the hdf node `metadata.keyspace`.  All of that logic is hidden away in the artifact builder code in a separate library.  We should probably consider, I don't know, a class method constructor to generate a new, empty artifact?  Ideas? @michaella21 @kfwilson

Reply:
yeah, I agree that it has a problem but I also didn't have any good idea how to handle this. There may be some way to handle the location inside of Artifact class but if we want to include versions, (and furthermore the list of components), we need a bridge between builder and Artifact class, maybe a classmethod factory, or any other idea?